dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/MultiDestConditionRouter.java

```diff
@@ -0,0 +1,422 @@
+import org.apache.dubbo.common.logger.LoggerFactory;
+import org.apache.dubbo.common.utils.CollectionUtils;
+import org.apache.dubbo.common.utils.Holder;
+import org.apache.dubbo.common.utils.NetUtils;
+import org.apache.dubbo.common.utils.StringUtils;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.RpcException;
+import org.apache.dubbo.rpc.cluster.router.RouterSnapshotNode;
+import org.apache.dubbo.rpc.cluster.router.condition.config.model.ConditionSubSet;
+import org.apache.dubbo.rpc.cluster.router.condition.config.model.DestinationSet;
+import org.apache.dubbo.rpc.cluster.router.condition.config.model.MultiDestCondition;
+import org.apache.dubbo.rpc.cluster.router.condition.matcher.ConditionMatcher;
+import org.apache.dubbo.rpc.cluster.router.condition.matcher.ConditionMatcherFactory;
+import org.apache.dubbo.rpc.cluster.router.state.AbstractStateRouter;
+import org.apache.dubbo.rpc.cluster.router.state.BitList;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.apache.dubbo.common.constants.LoggerCodeConstants.CLUSTER_CONDITIONAL_ROUTE_LIST_EMPTY;
+import static org.apache.dubbo.common.constants.LoggerCodeConstants.CLUSTER_FAILED_EXEC_CONDITION_ROUTER;
+import static org.apache.dubbo.rpc.cluster.Constants.DefaultRouteConditionSubSetWeight;
+import static org.apache.dubbo.rpc.cluster.Constants.RULE_KEY;
+
+public class MultiDestConditionRouter<T> extends AbstractStateRouter<T> {
+    public static final String NAME = "multi_condition";
+
+    private static final ErrorTypeAwareLogger logger = LoggerFactory.getErrorTypeAwareLogger(AbstractStateRouter.class);
+    protected static final Pattern ROUTE_PATTERN = Pattern.compile("([&!=,]*)\\s*([^&!=,\\s]+)");
+    private Map<String, ConditionMatcher> whenCondition;
+    private boolean trafficDisable;
+    private List<ConditionSubSet> thenCondition;
+    private int ratio;
+    private int priority;
+    private boolean force;
+    private boolean scopFroce;
+    protected List<ConditionMatcherFactory> matcherFactories;
+    private boolean enabled;
+
+    public MultiDestConditionRouter(
+            URL url, MultiDestCondition multiDestCondition, boolean enabled, boolean scopFroce) {
+        super(url);
+        this.scopFroce = scopFroce;
+        this.enabled = enabled;
+        matcherFactories =
+                moduleModel.getExtensionLoader(ConditionMatcherFactory.class).getActivateExtensions();
+        this.covert(multiDestCondition, this);
+        this.init(multiDestCondition.getFrom(), multiDestCondition.getTo());
+    }
+
+    public void init(Map<String, String> from, List<Map<String, String>> to) {
+        if (from == null || to == null) {
+            throw new IllegalArgumentException("Illegal route rule!");
```

Comment:
Make sure this exception will not affect RPC call execution. A warning log and skip should be fine.

Reply:
Here I keep the same logic as the old condition class(`ConditionStateRouter.class`), and I have the same try catch handling in the method caller ListenableStateRouter#process